src/main/kotlin/app/revanced/patches/youtube/utils/settings/SettingsPatch.kt

```diff
@@ -38,6 +42,49 @@ object SettingsPatch : BaseResourcePatch(
     requiresIntegrations = true
 ), Closeable {
 
+    private const val DEFAULT_ELEMENT = "About"
+    private const val DEFAULT_NAME = "ReVanced Extended"
+
+    private val SETTINGS_ELEMENTS_MAP = mapOf(
+        "Parent settings" to "@string/parent_tools_key",
+        "General" to "@string/general_key",
+        "Account" to "@string/account_switcher_key",
+        "Data saving" to "@string/data_saving_settings_key",
+        "Autoplay" to "@string/auto_play_key",
+        "Video quality preferences" to "@string/video_quality_settings_key",
+        "Background" to "@string/offline_key",
+        "Watch on TV" to "@string/pair_with_tv_key",
+        "Manage all history" to "@string/history_key",
+        "Your data in YouTube" to "@string/your_data_key",
+        "Privacy" to "@string/privacy_key",
+        "History & privacy" to "@string/privacy_key",
+        "Try experimental new features" to "@string/premium_early_access_browse_page_key",
+        "Purchases and memberships" to "@string/subscription_product_setting_key",
+        "Billing & payments" to "@string/billing_and_payment_key",
+        "Billing and payments" to "@string/billing_and_payment_key",
+        "Notifications" to "@string/notification_key",
+        "Connected apps" to "@string/connected_accounts_browse_page_key",
+        "Live chat" to "@string/live_chat_key",
+        "Captions" to "@string/captions_key",
+        "Accessibility" to "@string/accessibility_settings_key",
+        DEFAULT_ELEMENT to "@string/about_key"
+    )
+
+    private val InsertPosition by stringPatchOption(
+        key = "InsertPosition",
+        default = DEFAULT_ELEMENT,
+        values = SETTINGS_ELEMENTS_MAP,
+        title = "Insert position",
+        description = "Specify the setting name before which the RVX setting should be inserted."
+    )
+
+    private val CustomName by stringPatchOption(
```

Comment:
in my opinion it doesn't impact performance at all. on the fact that it can increase fictitious apps.. yeah, a bit. like all branding patches tho. it takes a second to change headers, icons and appicons etc

Reply:
Well, it's a customization options, why should we restrict people, right? 

Given that most of the people won't change the name it will break out of "heavy" code algorithm before it starts. 

But let's define, what's "heavy" and why it won't really cause any performance issues. 

This code goes through each language file (L) and each `<string>` element (S) inside those language files, so in worst case scenario it will be O(L×S) which can be considered as O(n•n) or O(n²). You probably already know that it's somewhere in the middle of Big O. But take a note that it is the worst case scenario. 

What we have in reality? "revanced_extended_settings_title" goes somewhere in the beginning of each language file, so it's even more faster, for the best case scenario it will be O(L). You won't notice any performance issues at all, it'll change the name in milliseconds even on slow devices.